packages/ui/src/utils/country-code.ts

```diff
@@ -0,0 +1,80 @@
+import i18next from 'i18next';
+import {
+  getCountries,
+  CountryCode,
+  CountryCallingCode,
+  getCountryCallingCode,
+} from 'libphonenumber-js';
+
+export const fallbackCountryCode = 'US';
+
+export const countryCallingCodeMap: Record<string, CountryCode> = {
+  zh: 'CN',
+  en: 'US',
+};
+
+export const isValidCountryCode = (countryCode: string): countryCode is CountryCode => {
+  try {
+    // Use getCountryCallingCode method to guard the input's value is in CountryCode union type, if type not match exceptions are expected
+    getCountryCallingCode(countryCode as CountryCode);
+
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const getDefaultCountryCode = (): CountryCode => {
+  const { language } = i18next;
+
+  // Extract the country code from language tag suffix
+  const [, countryCode] = language.split('-');
+
+  if (countryCode && isValidCountryCode(countryCode)) {
+    return countryCode;
+  }
+
+  return countryCallingCodeMap[language] ?? fallbackCountryCode;
+};
+
+export const getDefaultCountryCallingCode = () => getCountryCallingCode(getDefaultCountryCode());
+
+/**
+ * Provide Country Code Options
+ */
+export type CountryMetaData = {
+  countryCode: CountryCode;
+  countryCallingCode: CountryCallingCode;
+};
+
+export const getCountryList = (): CountryMetaData[] => {
+  const defaultCountryCode = getDefaultCountryCode();
+  const defaultCountryCallingCode = getCountryCallingCode(defaultCountryCode);
+
+  const countryList = getCountries()
+    .map((code) => ({
+      countryCode: code,
+      countryCallingCode: getCountryCallingCode(code),
+    }))
+    // Filter the detected default countryCode & duplicates
+    .filter(({ countryCallingCode }, index, self) => {
+      if (countryCallingCode === defaultCountryCallingCode) {
+        return false;
+      }
+
+      return (
+        self.findIndex((element) => element.countryCallingCode === countryCallingCode) === index
+      );
+    })
+    .slice()
+    // Sort by countryCallingCode
+    .sort((previous, next) => (next.countryCallingCode > previous.countryCallingCode ? -1 : 1));
```

Comment:
nit: previous.countryCallingCode - next.countryCallingCode

Reply:
just realize it's number. +1 for this

Reply:
string type not allowed